### PR TITLE
RHOAIENG-2510: Add test to validate the recreation of the KnativeServing resource

### PR DESCRIPTION
**Pull request by CFSNM**
Related ticket: https://issues.redhat.com/browse/RHOAIENG-2510

### CODE REVIEW

File: ods_ci/tests/Tests/100__deploy/130__operators/140__serverless_operator/140__serverless_operator.robot [hunk 1]
```diff
@@ -36,6 +36,22 @@ Validate DSC creates all Serverless CRs
...
+    ${return_code}    ${output}    Run And Return Rc And Output
+    ...    oc delete KnativeServing knative-serving -n ${KNATIVESERVING_NS} --ignore-not-found
+    Should Be Equal As Integers  ${return_code}   0   msg=Error deleting knative-serving, ${output}
+    Wait Until Keyword Succeeds    2 min    0 sec
+    ...        Check Number Of Resource Instances Equals To      KnativeServing     knative-serving      0
+    Wait Until Keyword Succeeds    6 min    0 sec
+    ...        Resource Should Exist     KnativeServing    knative-serving     ${KNATIVESERVING_NS}
+    Wait Until Keyword Succeeds    6 min    0 sec
+    ...        Resource Status Should Be     oc get KnativeServing knative-serving -n ${KNATIVESERVING_NS} -o json | jq '.status.conditions[] | select(.type=="Ready") | .status'     KnativeServing    "True"    # robocop: disable
+    Wait For Pods Numbers  2    namespace=${KNATIVESERVING_NS}
```
Comment (bdattoma) [inline ods_ci/tests/Tests/100__deploy/130__operators/140__serverless_operator/140__serverless_operator.robot:52]: there are usually more than 2 pods in that namespace, you sure this check won't be flaky?

Comment (CFSNM) [reply ods_ci/tests/Tests/100__deploy/130__operators/140__serverless_operator/140__serverless_operator.robot:52]: there are only 2 pods with the label "app.kubernetes.io/component=controller". I can add more conditions for the rest of pods with different labels, but this condition should be accurate

Comment (bdattoma) [reply ods_ci/tests/Tests/100__deploy/130__operators/140__serverless_operator/140__serverless_operator.robot:52]: ops, I missed the label selector. 
hmm maybe you could check the total pod number of that NS as well? maybe there is a common label selector

Comment (CFSNM) [reply ods_ci/tests/Tests/100__deploy/130__operators/140__serverless_operator/140__serverless_operator.robot:52]: sure, can try that

**Comment by CFSNM** [comment]
This test case is not a valid scenario

**Pull request closed without merging**